Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -69,15 +74,24 @@ class Navigator {
     static let shared: Navigator = Navigator()
     
     /**
-     Restrict direct initializer access.
+     Destroys and creates new instance of Navigator together with other related entities.
+     
+     Typically, this method is used to restart a Navigator with a specific Version during switching to offline or online modes.
+     - parameter version: String representing a tile version name. `nil` value means "latest". Specifying exact version also enables `fallback` mode which will passively monitor newer version available and will notify `tileVersionState` if found.
      */
-    private init() {
+    func restartNavigator(forcing version: String? = nil) {
```

Comment:
Basically, such an approach for recreating a Navigator might be a good candidate for making a static activation method somewhere so that we can stop putting initialization of the static fields of this class everywhere. Or at least we might consider making a struct with these fields and provide both an old approach with putting this activation config to existing places and a new approach with a public static activation method.

Reply:
I am not sure to follow the idea about this activation method. Do you mean to store all settings in a special struct and pass it to constructor/restart method? What are the advantages?

Reply:
An activation method is just a suggestion to think about.

> Do you mean to store all settings in a special struct and pass it to constructor/restart method? What are the advantages?

No, I basically mean the suggestion in this comment: https://github.com/mapbox/mapbox-navigation-ios/pull/3014#discussion_r635116232. So that we don't have to force-unwrap all fields, but to have a single field with all the parameters of an active navigator.

Reply:
See `NativeHandlersFactory` if I got it right (:

Reply:
This indeed looks great. Would you mind considering storing the object `factory` in `Navigator` directly and use dynamic properties for `historyRecorder`, `navigator` etc? In this case `factory` could be renamed to something else and creation for this factory might be moved to a private static method. This would allow us to reduce code duplication in `init` and `restartNavigator` to a minimum to prevent possible inconsistency in those methods.

Reply:
I thought it would be better to not store this factory because navigator settings could be potentially updated and if factory remains the same we either have to update it too (which adds more code) or such settings will be ignored on navigator restart.

Reply:
If it's potentially possible that only some of the parameters may change, then OK, the current approach is great